libs/wingc/src/jsify.rs

```diff
@@ -406,7 +406,11 @@ impl<'a> JSifier<'a> {
 					"{{\n{}}}\n",
 					fields
 						.iter()
-						.map(|(name, expr)| format!("\"{}\": {},", name.name, self.jsify_expression(expr, phase)))
+						.map(|(name, expr)| format!(
+							"\"{}\": {},",
+							snake_case_to_camel_case(&name.name),
```

Comment:
I think we can't automatically assume it's ok to do this conversion. We need to check the `evaluated_type` in `expression` and see that this struct type has `should_case_convert_jsii` set to `true`. This will probably require adding this field to the `Struct` struct in type_check.rs and fixing the JSII importer accordingly. If this all sounds too complex, let me know and I'll try to update this PR accordingly.

Reply:
@yoav-steinberg @marciocadev Please take a look at this: https://github.com/winglang/wing/issues/1049

Reply:
> @yoav-steinberg @marciocadev Please take a look at this: #1049

I commented in that issue. Lets resolve #1049 first and then continue from there.